ggshield/cmd/secret/scan/prereceive.py

```diff
@@ -195,35 +205,28 @@ def prereceive_cmd(
     if config.verbose:
         click.echo(f"Commits to scan: {len(commit_list)}", err=True)
 
-    try:
-        with ExitAfter(get_prereceive_timeout()):
-
-            scan_context = ScanContext(
-                scan_mode=ScanMode.PRE_RECEIVE,
-                command_path=ctx.command_path,
-            )
+    receiver, sender = multiprocessing.Pipe()
+    process = multiprocessing.Process(
+        target=_execute_prereceive,
+        args=(
+            config,
+            output_handler,
+            commit_list,
+            ctx.command_path,
+            ctx.obj["client"],
+            ctx.obj["exclusion_regexes"],
+            sender,
+        ),
+    )
 
-            return_code = scan_commit_range(
-                client=ctx.obj["client"],
-                cache=ReadOnlyCache(),
-                commit_list=commit_list,
-                output_handler=output_handler,
-                exclusion_regexes=ctx.obj["exclusion_regexes"],
-                matches_ignore=config.secret.ignored_matches,
-                scan_context=scan_context,
-                ignored_detectors=config.secret.ignored_detectors,
-            )
-            if return_code:
-                click.echo(
-                    remediation_message(
-                        remediation_steps=REMEDIATION_MESSAGE,
-                        bypass_message=BYPASS_MESSAGE,
-                        rewrite_git_history=True,
-                    ),
-                    err=True,
-                )
-            return return_code
-    except TimeoutError:
+    process.start()
+    process.join(timeout=get_prereceive_timeout())
+    if process.is_alive() or process.exitcode is None:
+        display_error("\nPre-receive hook took too long")
+        process.kill()
         return 0
-    except Exception as error:
-        return handle_exception(error, config.verbose)
+
+    if process.exitcode == 0:
+        return_code: int = receiver.recv()
+        return return_code
+    return process.exitcode
```

Comment:
If I understand the multiprocess.Process doc correctly, we should be able to pass the exit code via the child process exit code: make the child finish with `sys.exit(return_code)` and forward it here.

This would simplify the code (and the tests) by removing the need for the Pipe. Have you tried this?